src/java/org/apache/cassandra/serializers/UserTypeSerializer.java

```diff
@@ -56,8 +57,14 @@ public void validate(ByteBuffer bytes) throws MarshalException
                 throw new MarshalException(String.format("Not enough bytes to read %dth field %s", i, entry.getKey()));
 
             ByteBuffer field = ByteBufferUtil.readBytes(input, size);
-            entry.getValue().validate(field);
-            i++;
```

Comment:
@dcapwell So it was possible for this increment to get eaten up by the `continue` above?

Reply:
yep, continue caused every error to be wrong.